homeassistant/components/unifi/sensor.py

```diff
@@ -169,6 +179,11 @@ class UnifiSensorEntityDescription(
     """Class describing UniFi sensor entity."""
 
     is_connected_fn: Callable[[UniFiController, str], bool] | None = None
+    # Custom function to determine whether a state change should be recorded
+    value_changed_fn: Callable[
```

Comment:
Lets make it optional so the value default to None instead, no need to call the lambda every time.

Reply:
If I make the default `None`, then the call site in `async_update_state` would now need to change from this:

```python
        # Update the value only if value is considered to have changed relative to its previous state
        if description.value_changed_fn(
            self.native_value, (value := description.value_fn(self.controller, obj))
        ):
            self._attr_native_value = value
```

to something like this (to re-instate the basic `!=` check that needs to be there when this fn isn't overridden):

```python
        # Update the value only if value is considered to have changed relative to its previous state
        new_value = description.value_fn(self.controller, obj)
        if (description.value_changed_fn is not None and description.value_changed_fn(self.native_value, new_value)) or self._attr_native_value != new_value:
            self._attr_native_value = new_value
```

I chose to treat the original `if old != new` condition as the default `value_changed` function to make the execution flow a little simpler. Is the function call performance cost the concern? (I understood this isn't a huge concern in newer versions of Python, but I haven't done benchmarks. We already lean on callables a lot already in entity implementations, so I figured generalizing these concepts would be more idiomatic.)

Reply:
Thank you for the review! I'll let the latest revision run in my dev container for a while to watch how it holds up.

Reply:
Sounds good!